src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,34 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_TIME_FORMATTER =
+            DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder resultStr = new StringBuilder();
+        resultStr.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        for (String name : names) {
+            resultStr.append(System.lineSeparator()).append(name).append(" - ");
+            int sum = 0;
+            for (String d : data) {
+                String[] splitData = d.split(" ");
+                if (splitData[1].equals(name) && isRelevantDate(splitData[0], dateFrom, dateTo)) {
+                    sum += Integer.parseInt(splitData[2]) * Integer.parseInt(splitData[3]);
+                }
+            }
```

Comment:
let's make you code more readable

first of all I would rename variable `sum` to `salary` since name `sum` is way too general (sum of what?)

also I suggest creating a separate constant for personal data delimiter like :
```
private static final String PERSONAL_DATA_DELIMITER = " ";
```

apart from that, you need to rename the `d` parameter in `for` loop to something like `personData` or `personalData`. after that you should declare separate variables for each particular column, like:
```
String workingDate = splitPersonalData[0];
String personName = splitPersonalData[1];
String workingHours = splitPersonalData[2];
String salarayPerHour = splitPersonalData[3];
```

also I would extract salary calculation to a separate method like:
```
private int calculateDailySalary(String workingHours, String salaryPerHour) {
   return Integer.parseInt(workingHours) * Integer.parseInt(salaryPerHour);
}
```

```suggestion
            int salary = 0;
            for (String personalData : data) {
                String[] splitPersonalData= d.split(PERSONAL_DATA_DELIMITER);
                String workingDate = splitPersonalData[0];
                String personName = splitPersonalData[1];
                String workingHours = splitPersonalData[2];
                String salarayPerHour = splitPersonalData[3];

                if (personName.equals(name) && isRelevantDate(workingDate, dateFrom, dateTo)) {
                    salary += calculateDailySalary(workingHours, salarayPerHour);
                }
            }
```

Reply:
@teract10s you haven't created `calculateDailySalary` method :)